### PR TITLE
Use pyside2-rcc if pyrcc5 fail

### PR DESCRIPTION
**Pull request by Czaki**
# Description

Re implement of #1080 which was broken in #1262, this time with test
https://github.com/napari/napari/issues/1618#issuecomment-687473278

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)

# References

# How has this been tested?

- [x] example: all tests pass with my change

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by Czaki** [comment]
I do not know how to create windows executable and have no experience with this so I disable test for windows. 

@sofroniewn ready for review

File: napari/resources/test_resources.py [hunk 1]
```diff
@@ -1,9 +1,36 @@
...
+        f_p.write(f"echo Fake {file_name} executable\n")
+        f_p.write("exit 10")
+
+    (tmp_path / "path" / file_name).chmod(0o777)
+
+    monkeypatch.setenv(
+        "PATH", os.pathsep.join([str(tmp_path / "path"), os.environ["PATH"]])
+    )
+    build_pyqt_resources(str(tmp_path / "resourcses.py"))
+    assert (tmp_path / "resourcses.py").exists()
```
Comment (tlambert03) [inline napari/resources/test_resources.py]:
```suggestion
    build_pyqt_resources(str(tmp_path / "resources.py"))
    assert (tmp_path / "resources.py").exists()
```

File: napari/resources/test_resources.py [hunk 2]
```diff
@@ -1,9 +1,36 @@
...
 
 
 def test_resources():
     """Test that we can build icons and resources."""
     out = import_resources(version='test')
     os.remove(out)
+
+
+def test_path_pollution(tmp_path, monkeypatch, capfd):
+    if qtpy.API == "pyqt5":
```
Comment (tlambert03) [inline napari/resources/test_resources.py:24]: it's not entirely clear to me what cases you're catching with this test.  For instance, this passes for me (locally) already on master, without this PR.  Can you add a docstring to this test explaining the purpose of it?

**Review by tlambert03 — APPROVED** [review]

File: napari/resources/test_resources.py (hunk 2 above)
Comment (Czaki) [reply napari/resources/test_resources.py:24]: Did you use pyqt5 as backend?

Comment (tlambert03) [reply napari/resources/test_resources.py:24]: yeah

Comment (Czaki) [reply napari/resources/test_resources.py:24]: So this test, in old version properly, work on pyside2 environment.
Current version should fail on master. 

Looks better now?

**Review by sofroniewn — APPROVED** [review]
Looks great, @Czaki, thanks for the fix and test. I will merge now!

**Pull request merged**